plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/functions/tablechanges/TableChangesFunction.java

```diff
@@ -100,9 +105,38 @@ public TableFunctionAnalysis analyze(
         long sinceVersion = -1; // -1 to start from 0 when since_version is not provided
         if (sinceVersionValue != null) {
             sinceVersion = (long) sinceVersionValue;
-            checkFunctionArgument(sinceVersion >= 0, "Invalid value of since_version: %s. It must not be negative.", sinceVersion);
+            checkFunctionArgument(sinceVersion >= 0 || sinceVersion == DEFAULT_GUARD_VALUE, "Invalid value of since_version: %s. It must not be negative.", sinceVersion);
+        }
+
+        ScalarArgument sinceTimestampArgument = (ScalarArgument) arguments.get(SINCE_TIMESTAMP_ARGUMENT);
+        Object sinceTimestampValue = sinceTimestampArgument.getValue();
+        long sinceTimestamp = 0L;
+        if (sinceTimestampValue != null) {
+            sinceTimestamp = ((Long) sinceTimestampValue) / MICROSECONDS_PER_MILLISECOND;
+            checkFunctionArgument(sinceTimestamp > 0, "Invalid value of since_timestamp: %s. It must not be negative.", sinceTimestamp);
+        }
+
+        if (sinceVersionValue == null && sinceTimestampValue != null) { // to prevent system.table_changes('schemaName', 'tableName', null, TIMESTAMP '2024-10-31 01:00')
```

Comment:
put the comment inside the IF on its own line (omitting initial "to")
document also "Why" of this. What made you think it's important to prevent such usages.